main.c

```diff
@@ -35,37 +31,26 @@ void selectionSort(int a[], int size)
 
 int main(int argc, char *argv[])
 {
-    int arr[argc-1];
-    int i = 1;
-    char *p;
-    long longvalue;
-    int value;
+    size_t size = argc - 1; // size of array
+    int *arr = malloc(size * sizeof(int)); // heap-allocated array of ints
 
     /* Putting the command line args
      * as ints in the array */
 
-    while(--argc > 0)
-    {
-        longvalue = strtol(argv[i], &p, 10);
-        value = (int) longvalue;
-        arr[i-1] = value;
-        i++;
+    for (size_t i = 0; i < size; ++i) {
+        arr[i] = atoi(argv[i + 1]);
     }
 
-    size_t s2 = sizeof(arr) / sizeof(arr[0]);
-    int size2 = (int) s2;
+    printf("Input Array: \n");
```

Comment:
a sorting function should not print anything, it should just short. Printing should be done by the caller of sorting function.